lib/commons/color/color.js

```diff
@@ -1,68 +1,7 @@
 import standards from '../../standards';
 
-/**
- * Convert a CSS color value into a number
- */
-function convertColorVal(colorFunc, value, index) {
-  if (/%$/.test(value)) {
-    //<percentage>
-    if (index === 3) {
-      // alpha
-      return parseFloat(value) / 100;
-    }
-    return (parseFloat(value) * 255) / 100;
-  }
-  if (colorFunc[index] === 'h') {
-    // hue
-    if (/turn$/.test(value)) {
-      return parseFloat(value) * 360;
-    }
-    if (/rad$/.test(value)) {
-      return parseFloat(value) * 57.3;
-    }
-  }
-  return parseFloat(value);
-}
-
-/**
- * Convert HSL to RGB
- */
-function hslToRgb([hue, saturation, lightness, alpha]) {
-  // Must be fractions of 1
-  saturation /= 255;
-  lightness /= 255;
-
-  const high = (1 - Math.abs(2 * lightness - 1)) * saturation;
-  const low = high * (1 - Math.abs(((hue / 60) % 2) - 1));
-  const base = lightness - high / 2;
-
-  let colors;
-  if (hue < 60) {
-    // red - yellow
-    colors = [high, low, 0];
-  } else if (hue < 120) {
-    // yellow - green
-    colors = [low, high, 0];
-  } else if (hue < 180) {
-    // green - cyan
-    colors = [0, high, low];
-  } else if (hue < 240) {
-    // cyan - blue
-    colors = [0, low, high];
-  } else if (hue < 300) {
-    // blue - purple
-    colors = [low, 0, high];
-  } else {
-    // purple - red
-    colors = [high, 0, low];
-  }
-
-  return colors
-    .map(color => {
-      return Math.round((color + base) * 255);
-    })
-    .concat(alpha);
-}
+const hexRegex = /^#[0-9a-f]{3,8}$/i;
```

Comment:
I took this opportunity to do something I've been wanting to do for awhile now which is convert the `Color` constructor function into a true class. The advantage to this was that I was able to do `deepEquals` on two Color objects whereas before that was not possible (thus greatly simplifying the tests). Another advantage is that all Color objects now share memory for all their functions thanks to the prototype whereas before each Color object had their own memory instance for each function.